platforms/jvm/plugins-application/src/main/java/org/gradle/api/internal/plugins/StartScriptTemplateBindingFactory.java

```diff
@@ -188,7 +237,25 @@ String createJoinedAppHomeRelativePath(String scriptRelPath) {
             appHomeRelativePath.add("..");
         }
 
-        return Joiner.on(windows ? "\\" : "/").join(appHomeRelativePath);
+        return Joiner.on(getPathElementSeparator()).join(appHomeRelativePath);
+    }
+
+    /**
+     * The separator used to separate each element in a path.
+     *
+     * @return the path element separator
+     */
+    private String getPathElementSeparator() {
+        return windows ? "\\" : "/";
```

Comment:
I think you can use `File.separator`

Reply:
No, we generate both scripts on one OS - e.g., if you do `gradle installDist`, it has both Windows and Unix. So we need our own logic.